src/main/java/dk/cngroup/lentils/controller/HintController.java

```diff
@@ -63,48 +73,74 @@ public String deleteHint(@PathVariable("hintId") final Long hintId) {
 
     @GetMapping(value = "/update/{hintId}")
     public String updateHintForm(@PathVariable("hintId") final Long hintId, final Model model) {
-        HintFormDTO hintFormDto = mapper.map(hintService.getHint(hintId), HintFormDTO.class);
+        Hint hint = hintService.getHint(hintId);
+        HintFormDTO hintFormDto = mapper.map(hint, HintFormDTO.class);
         model.addAttribute(TEMPLATE_ATTR_HEADING, HEADING_EDIT_HINT);
         model.addAttribute(TEMPLATE_ATTR_HINT, hintFormDto);
+        model.addAttribute(TEMPLATE_ATTR_FILENAME, imageService.getFileNameFromEntity(hint));
+        if (hintFormDto.getImage() == null) {
+            model.addAttribute(TEMPLATE_ATTR_FILE, "");
+        } else {
+            model.addAttribute(TEMPLATE_ATTR_FILE, MvcUriComponentsBuilder.fromMethodName(HintController.class,
+                    "serveFile", hintFormDto.getImage().getOriginalFilename()).build().toString());
+        }
         return VIEW_ADMIN_HINT_DETAIL;
     }
 
+    @GetMapping("/files/{filename:.+}")
+    public ResponseEntity<Resource> serveFile(final @PathVariable String filename) {
+        Resource file = imageService.loadAsResource(imageService.getFilePathToShow(filename));
+        return ResponseEntity.ok().header(HttpHeaders.CONTENT_DISPOSITION,
+                "attachment; filename=\"" + file.getFilename() + "\"").body(file);
+    }
+
     @PostMapping(value = "/update/{hintId}")
     public String updateHint(@PathVariable("hintId") final Long hintId,
                        @Valid @ModelAttribute(TEMPLATE_ATTR_HINT) final HintFormDTO hintFormDto,
                        final BindingResult bindingResult,
                        final Model model) {
+        Hint hint = hintService.getHint(hintId);
+        imageService.setImageToFormObject(hint, hintFormDto);
```

Comment:
OPTIONAL: Muze byt metoda `setImageToFormObject` vytvorena jako staticka metoda? Pokud nepotrebuje dalsi zavoslosti, pak bych to tak mozna udelal. Samozrejme i s dalsimi pomocnymi metodami jako `getFile` atd. Ale opravdu jen v pripade, ze nepotrebuje dalsi zavislosti (pro ukladani souboru, nacitani, ostatni setvices atd), jinak to samozrejme nepujde. Pokud ano, vytvorich bych pro to tridu napr. "FileUtils"